pyserial dependency for hal_espressif

(previously https://github.com/zephyrproject-rtos/hal_espressif/issues/203)

Targeting an Espressif `esp32` requires `pyserial` to be installed. However, it will only get installed if a user uses `requirements-run-test.txt ` (not common) or gets ALL dependencies using `requirements.txt` per the Getting Started Guide. However, since `pyserial` isn't included in `requirements-base.txt`, anyone trying to maintain a lean build environment (like CI) will run into a dependency issue.

I came across this while developing a [Docker](https://github.com/beriberikix/zephyr-docker/blob/main/Dockerfile) [container ](https://github.com/wokwi/wokwi-builders/blob/main/zephyr-esp32/Dockerfile) but the steps are reproducible on any machine, such as a Linux host.

```
# Assume Zephyr OS dependencies & SDK are installed
pip3 install west
pip3 install -r https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/v3.3.0/scripts/requirements-base.txt
pip3 install requests // #56215
west init ~/zephyrproject
cd zephyrproject
west update
west blobs  fetch hal_espressif
cd zephyr
west build -b esp32 samples/basic/blinky -p
```

The error will be something like:

```
FAILED: .bin_timestamp /home/wokwi/zephyrproject/zephyr/build/esp-idf/build/bootloader/.bin_timestamp 
#0 17.95 cd /home/wokwi/zephyrproject/zephyr/build/esp-idf/build/bootloader && /usr/bin/python3.9 /home/wokwi/zephyrproject/modules/hal/espressif/components/esptool_py/esptool/esptool.py --chip esp32 elf2image --flash_mode dio --flash_freq 40m --flash_size 2MB --dont-append-digest -o /home/wokwi/zephyrproject/zephyr/build/esp-idf/build/bootloader/bootloader.bin /home/wokwi/zephyrproject/zephyr/build/esp-idf/build/bootloader/bootloader.elf && /usr/local/lib/python3.9/dist-packages/cmake/data/bin/cmake -E echo "Generated /home/wokwi/zephyrproject/zephyr/build/esp-idf/build/bootloader/bootloader.bin" && /usr/local/lib/python3.9/dist-packages/cmake/data/bin/cmake -E md5sum /home/wokwi/zephyrproject/zephyr/build/esp-idf/build/bootloader/bootloader.bin > /home/wokwi/zephyrproject/zephyr/build/esp-idf/build/bootloader/.bin_timestamp
#0 17.95 Pyserial is not installed for /usr/bin/python3.9. Check the README for installation instructions.
#0 17.95 Traceback (most recent call last):
#0 17.95   File "/home/wokwi/zephyrproject/modules/hal/espressif/components/esptool_py/esptool/esptool.py", line 31, in <module>
#0 17.95     import esptool
#0 17.95   File "/home/wokwi/zephyrproject/modules/hal/espressif/components/esptool_py/esptool/esptool/__init__.py", line 41, in <module>
#0 17.95     from esptool.cmds import (
#0 17.95   File "/home/wokwi/zephyrproject/modules/hal/espressif/components/esptool_py/esptool/esptool/cmds.py", line 14, in <module>
#0 17.95     from .bin_image import ELFFile, ImageSegment, LoadFirmwareImage
#0 17.95   File "/home/wokwi/zephyrproject/modules/hal/espressif/components/esptool_py/esptool/esptool/bin_image.py", line 14, in <module>
#0 17.95     from .loader import ESPLoader
#0 17.95   File "/home/wokwi/zephyrproject/modules/hal/espressif/components/esptool_py/esptool/esptool/loader.py", line 30, in <module>
#0 17.95     import serial
#0 17.95 ModuleNotFoundError: No module named 'serial'
#0 17.95 ninja: build stopped: subcommand failed.
#0 17.95 ninja: build stopped: subcommand failed.
#0 17.96 FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /home/wokwi/zephyrproject/zephyr/build
```

If I install `pyserial` before building, the build completes succesfully.

--- a/scripts/requirements-base.txt
+++ b/scripts/requirements-base.txt
@@ -19,6 +19,8 @@ packaging
 progress
 psutil
 pylink-square
+pyserial
+requests
 
 # for ram/rom reports
 anytree

--- a/scripts/requirements-run-test.txt
+++ b/scripts/requirements-run-test.txt
@@ -2,9 +2,6 @@
 #
 # things used by twister or related in run time testing
 
-# used to connect to boards for console IO
-pyserial
-
 # used to flash & debug various boards
 pyocd>=0.29.0
 

